Backend/web api/controllers

Nafanyan opened the pull request.

Nafanyan committed:
Начал делать контроллер

Nafanyan committed:
Сделал контроллер для User'а

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,101 @@
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetByIdAsync([FromRoute] long id)
+        {
+            GetUserByIdDto getUserByIdDto = new GetUserByIdDto
```
Почему бы нам не создавать тут квери сразу? Тут лишний совершенно объект

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,101 @@
+            {
+                Id = id
+            };
+            QueryResult<User> queryResult = await _getUserByIdQueryHandler.HandleAsync(getUserByIdDto.Map());
```
А где проверка что валидация прошла или не прошла? Думаю стоит не Ok() возвращать, а BadRequest если валидация не прошла)

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,101 @@
+            _unitOfWork = unitOfWork;
+        }
+
+        [HttpGet("{id}")]
```
userId

SherlockD replied:
хотяя, в целом и так понятно, давай оставим

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,101 @@
+        [HttpPost()]
+        public async Task<IActionResult> AddAsync([FromBody] CreateUserDto createUserDto)
+        {
+            CommandResult commandResult = await _createUserCommandHandler.HandleAsync(createUserDto.Map());
```
Вот тут да, согласен что мапить нужно

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,101 @@
+        }
+
+        [HttpPut("update-login")]
+        public async Task<IActionResult> PutLogin([FromBody] UpdateUserLoginDto updateUserLoginDto)
```
Update login

SherlockD replied:
не знаю конечно где пригодится этот метод, вроде как ни в одном сервисе логин нельзя поменять)

SherlockD commented on Web/Presentation.Intranet.Api/Mappers/UserMappers/CreateUserCommandMapper.cs:
```diff
@@ -0,0 +1,17 @@
+{
+    public static class CreateUserCommandMapper
+    {
+        public static CreateUserCommand Map(this CreateUserDto createUserDto)
```
молодец что использовал расширения, правильное решение

Nafanyan committed:
Сделал правки согласно последним замечаниям:
1. Добавил BadRequest
2. Убрал лишние сущности и мапперы
3. Подредактировал названия

Nafanyan committed:
update-password => Update-password

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,131 @@
+
+            if (queryResult.ValidationResult.IsFail)
+            {
+                return BadRequest(queryResult.ValidationResult.Error);
```
тут можно также query result возвращать

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,131 @@
+        }
+
+        [HttpPut("Update-password")]
+        public async Task<IActionResult> PutPassword([FromBody] UpdateUserPasswordDto updateUserPasswordDto)
```
UpdatePassword, ты ведь при post запросе не пишешь PostCreateUser) это не правильное решение

Nafanyan committed:
создал контроллер Event'ов

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/EventsController.cs:
```diff
@@ -0,0 +1,95 @@
+namespace Presentation.Intranet.Api.Controllers
+{
+    [ApiController]
+    [Route("Users")]
```
Почему Users

SherlockD replied:
А, я понял, так то да, норм, мы же через юзера туда обращаемся.

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/EventsController.cs:
```diff
@@ -0,0 +1,95 @@
+            _unitOfWork = unitOfWork;
+        }
+
+        [HttpGet("{userId}/[controller]/{startEvent}/{endEvent}")]
```
startEvent, endEvent как будто бы гет параметрыми нужно сделать, не понятно из пути что там за поля

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/EventsController.cs:
```diff
@@ -0,0 +1,95 @@
+        }
+
+        [HttpPost("{userId}/[controller]")]
+        public async Task<IActionResult> CreateEvent([FromBody] CreateEventDto createEventDto)
```
Предлагаю переименовать Dto->Request

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/EventsController.cs:
```diff
@@ -0,0 +1,95 @@
+        [HttpPost("{userId}/[controller]")]
+        public async Task<IActionResult> CreateEvent([FromBody] CreateEventDto createEventDto)
+        {
+            CommandResult commandResult = await _createEventCommandHandler.HandleAsync(createEventDto.Map());
```
у тебя в роуте уже есть userId, но ты нигде не исопльзуешь его, подумай как можно сделать

Nafanyan committed:
Настроил валидацию при вводе строк для даты

Nafanyan committed:
Исправил согласно последним замечаниям;
Поработал над валидацией

SherlockD commented on Web/Presentation.Intranet.Api/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,130 @@
+        }
+
+        [HttpPut("Update-login")]
+        public async Task<IActionResult> UpdateLogin([FromBody] UpdateUserLoginDto updateUserLoginDto)
```
А мы эти два метода не можем одним написать?) Типа UpdateUser(UserDto) и обновлять)

Nafanyan replied:
ну не, давай оставим так)

Чтобы обновить пароль нужно ввести старый и новый;
Чтобы обновить логин нужно ввести новый логин и действующий пароль.

Nafanyan committed:
Заменил DateTime во всех командах и запросах на string

Nafanyan committed:
начал использовать наследуемое свойство Entities

Nafanyan committed:
Исправил контроллеры в соответствии с правками в Application

Nafanyan committed:
Убрал валидацию из маппера ЕЕЕЕЕжжжжиии

SherlockD commented on Application/Events/Commands/CreateEvent/CreateEventCommandValidation.cs:
```diff
@@ -30,17 +30,30 @@ public async Task<ValidationResult> ValidationAsync(CreateEventCommand command)
 
-            if (command.StartEvent.ToShortDateString != command.EndEvent.ToShortDateString)
+            DateTime startEvent;
+            DateTime.TryParse(command.StartEvent, out startEvent);
```
подумай как исправить

Nafanyan committed:
Подправил контроллеры b bcghfdbk dfkblfnjhs Event'a

Nafanyan committed:
Проверил сваггер, убрал мелкие недочеты
